src/platform_specific/affinity.unix.cc

```diff
@@ -0,0 +1,29 @@
+
+#include <cassert>
+#include <cstdint>
+
+#include <pthread.h>
+#include <sched.h>
+
+#include "affinity.h"
+
+namespace celerity {
+namespace detail {
+
+	uint32_t affinity_counter(const cpu_set_t& base_mask) {
```

Comment:
I don't think this function is needed, there is a `CPU_COUNT` macro that does exactly this!